lib/copy_gauge_offset.cu

```diff
@@ -0,0 +1,127 @@
+#include <gauge_field.h>
+#include <gauge_field_order.h>
+
+#include <kernels/copy_field_offset.cuh>
+#include <instantiate.h>
+
+namespace quda
+{
+
+  template <class Field, class Element, class G> void copy_gauge_offset(Field &out, const Field &in, CommKey offset)
+  {
+    G out_accessor(out);
+    G in_accessor(in);
+    using Arg = CopyFieldOffsetArg<Field, Element, G>;
+    Arg arg(out_accessor, out, in_accessor, in, offset);
+    CopyFieldOffset<Arg> copier(arg, in);
+  }
+
+  template <typename Float, int nColor> struct CopyGaugeOffset {
+    CopyGaugeOffset(GaugeField &out, const GaugeField &in, CommKey offset)
```

Comment:
Can we add a separate piece in `instantiate.h` for this? Or is this so specialized there's no point in exposing it for other files?

Reply:
I will file and issue and do it later ...

Reply:
#1090.